client/core/account.go

```diff
@@ -2,11 +2,69 @@ package core
 
 import (
 	"encoding/hex"
+	"fmt"
 
 	"decred.org/dcrdex/client/db"
 	"github.com/decred/dcrd/dcrec/secp256k1/v3"
 )
 
+// AccountDisable is used to disable an account by given host and application
+// password.
+func (c *Core) AccountDisable(pw []byte, host string) error {
+	// Validate password.
+	_, err := c.encryptionKey(pw)
+	if err != nil {
+		return codedError(passwordErr, err)
+	}
+
+	// Parse address.
+	addr, err := addrHost(host)
+	if err != nil {
+		return newError(addressParseErr, "error parsing address: %v", err)
+	}
+
+	// Get the dexConnection.
+	c.connMtx.RLock()
+	defer c.connMtx.RUnlock()
+	dc, found := c.conns[addr]
+	if !found {
+		return newError(unknownDEXErr, "DEX: %s", addr)
+	}
+
+	// Check active orders.
+	if dc.hasActiveOrders() {
+		return fmt.Errorf("cannot disable account with active orders")
+	}
+
+	acctInfo, err := c.db.Account(dc.acct.host)
+	if err != nil {
+		return newError(accountRetrieveErr, "Error retrieving account: %v", err)
+	}
+	err = c.db.DisableAccount(acctInfo)
+	if err != nil {
+		return newError(accountDisableErr, "Error disabling account: %v", err)
+	}
+	// Stop dexConnection books.
+	dc.cfgMtx.RLock()
+	for _, m := range dc.cfg.Markets {
+		// Emptry bookie's feeds map & stop close timers.
+		dc.booksMtx.Lock()
+		if b, found := dc.books[m.Name]; found {
+			b.feeds = make(map[uint32]*BookFeed, 1)
```

Comment:
Shouldn't just discard the old feeds. I think you can iterate through `b.feeds` and `close` the `BookFeed.C` channels. Looks like we're handling that in websocket, but not in `loadbot`, which is fine for now since we won't be changing the app password there.

Reply:
We might to live with the websocket's `marketSyncer` erroring trying to unsubscribe because of the `defer m.feed.Close()` that is really a (very obfuscated with wrappers of wrappers) callback to `(*dexConnection).stopBook`  in `(*marketSyncer).Run`.  I was referring to this in my two comments [here](https://github.com/decred/dcrdex/pull/985#issuecomment-789832980).  One solution I can see is eliminating that defer and only having `(*marketSyncer).Run` do `m.feed.Close` on context cancellation, not a closed feed channel.  But @buck54321's suggestion is what I was saying with:

>  For consumers receiving on the BookFeed.C channel, it will also be necessary to close these channels after the removing the BookFeed from the feeds map.

Stopping the timer will prevent redundant unsubscribes from `marketSyncer`s that stopped before doing this disable function, but for active `marketSyncer`s, the channel close is needed so they can return.  I think making the `m.feed.Close` call in `(*marketSyncer).Run` not a defer could solve the issue of `stopBook` getting called when the channel is closed.

Reply:
To restate that, [`Run`](https://github.com/decred/dcrdex/blob/bea35a9e57536b4c3d3bd8215992710e89fae0eb/client/websocket/websocket.go#L239) returns in two ways:

1. `wsUnmarket` does `feedLoop.Stop`, which cancels the context
2. `update, ok := <-m.feed.C` where `!ok` (closed channel)

When you navigate away from the /markets page, such as when going to /settings, it does `wsUnmarket`.

But there can be other tabs or websocket consumers with active `marketSyncer`s (`Run`) going when you do disable account.  So to get them to return, gotta close that `feed.C` channel.

And because of how `m.feed.Close()` cascades all the way to `(*bookie).closeFeed`, it creates a new `closeTimer`, which in return can call `dexConnection.stopBook` and attempt to unsubscribe from long after loosing the connection to the dex server.

Reply:
With the closed channel handling in `(*marketSyncer).Run` getting sorted out, don't forget to actually close the channel so `(*marketSyncer).Run` does hit the closed channel `case`.  This was @buck54321's suggestion.  I just wanted to be clear about what closing the channel does in the receiver.